lib/loaders/file.js

```diff
@@ -23,12 +24,31 @@ module.exports = function (refValue, options) {
   }
 
   const filePath = getRefFilePath(refPath)
+  const filePathLowerCase = filePath.toLowerCase()
 
   let newValue
   try {
-    var data = fs.readFileSync(filePath)
-    newValue = JSON.parse(data)
-  } catch (e) {}
+    var data = fs.readFileSync(filePath, 'utf8')
+    if (filePathLowerCase.endsWith('.json')) {
+      newValue = parseJson(data) || parseYaml(data)
+    } else if (filePathLowerCase.endsWith('.yml') || filePathLowerCase.endsWith('.yaml')) {
+      newValue = parseYaml(data)
+    } else {
+      newValue = parseJson(data) || parseYaml(data)
```

Comment:
This seems redundant with the first case? Looks like the first case could just be the .yml/.yaml case and then this else?

Reply:
the intention of this is to handle files with no extension. the question is wether you'd like to support it or not.